tests/PatternAssertions/SocialMediaLinksAssert.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\PatternAssertions;
+
+use Symfony\Component\DomCrawler\Crawler;
+use PHPUnit\Framework\Exception;
+
+/**
+ * Assertions for the social media link pattern.
+ *
+ * @package Drupal\Tests\oe_theme\PatternAssertions
+ */
+class SocialMediaLinksAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'title' => [
+        [$this, 'assertElementText'],
+        'div' . $this->getBaseItemClass($variant) . ' p.ecl-social-media-follow__description',
+      ],
+      'links' => [
+        [$this, 'assertLinks'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+    $field_list_container = $crawler->filter('div' . $this->getBaseItemClass($variant));
+    self::assertCount(1, $field_list_container);
+  }
+
+  /**
+   * Returns the base CSS selector of a pattern depending on the variant.
+   *
+   * @param string $variant
+   *   The variant being checked.
+   *
+   * @return string
+   *   The base selector for the variant.
+   */
+  protected function getBaseItemClass(string $variant): string {
+    if ($variant === 'horizontal') {
+      return '.ecl-social-media-follow.ecl-social-media-follow--horizontal';
+    }
+    return '.ecl-social-media-follow.ecl-social-media-follow--vertical';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getPatternVariant(string $html): string {
+    $crawler = new Crawler($html);
+    $field_list_container = $crawler->filter('div.ecl-social-media-follow');
+    $existing_classes = $field_list_container->attr('class');
+    $existing_classes = explode(' ', $existing_classes);
+    if (in_array('ecl-social-media-follow--horizontal', $existing_classes)) {
+      return 'horizontal';
+    }
+    if (in_array('ecl-social-media-follow--vertical', $existing_classes)) {
+      return 'vertical';
+    }
+    throw new Exception('Variant of social media links pattern is not identified.');
+  }
+
+  /**
+   * Asserts the links of the pattern.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertLinks(array $expected_items, Crawler $crawler): void {
+    $li_items = $crawler->filter('li.ecl-social-media-follow__item');
+    self::assertCount(count($expected_items), $li_items);
+
+    foreach ($expected_items as $index => $expected_item) {
+      $li_item = $li_items->eq($index);
+      if (empty($expected_item['service'])) {
```

Comment:
Should we print the link if there is no service?
It's mandatory in ecl : https://ec.europa.eu/component-library/ec/components/social-media-follow/usage/
There is no validation in twig, but if there is no service maybe we shouldn't print the link at all.
This removed part from the deprecated tpl was building a link only if there was a service we should keep the same logic:
https://github.com/openeuropa/oe_theme/pull/633/files#diff-41b2bab11b31265a931f90314e29dd9656b15c21edc1383988bf29c834dd5e2bL14-L31
Which is in fact updated but if a link doesn't have a service is still printed:
<img width="406" alt="link without service" src="https://user-images.githubusercontent.com/22908988/101909172-caaf1500-3bc5-11eb-9e02-61008f7e85af.png">